drivers/interrupt_controller/intc_plic.c

```diff
@@ -21,15 +21,30 @@
 #include <zephyr/drivers/interrupt_controller/riscv_plic.h>
 #include <zephyr/irq.h>
 
+#define PLIC_BASE_ADDR(n) DT_INST_REG_ADDR(n)
+/*
+ * These registers' offset are defined in the RISCV PLIC specs, see:
+ * https://github.com/riscv/riscv-plic-spec
+ */
+#define PLIC_REG_PRIO_OFFSET 0x0
+#define PLIC_REG_IRQ_EN_OFFSET 0x2000
+#define PLIC_REG_REGS_OFFSET 0x200000
+/*
+ * Trigger type is mentioned, but not defined in the RISCV PLIC specs.
+ * However, it is defined and supported by at least the Andes & Telink datasheet, and supported
+ * in Linux's SiFive PLIC driver
+ */
+#define PLIC_REG_TRIG_TYPE_OFFSET 0x1080
```

Comment:
Hi @ycsin , @cfriedt , @fkokosinski 

Guys, sorry for my late comment on it. I've just stumbled across this hardcoded  offset during the migration to Zephyr 3.5.

I believe hardcoding this offset and making it a default for all the vendors and platforms is a mistake. 
Trigger Type register is not defined in the recent PLIC Spec and so is not compliant with RISC-V Privileged Standard: https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic-1.0.0.pdf
IMHO this is a miss in the Spec and it results in a situation when all the vendors who support Triggering type configuration implement it in a proprietary way, using the custom offsets and custom codes for the triggering types.

So until it becomes a standard, it would be nice if we leave a room for flexibility here. I think returning PLIC_SUPPORTS_EDGE_IRQ configuration could be a nice option. 

Hope to hear from you. Thanks!

Reply:
Hi @ycsin , @cfriedt , @fkokosinski , 

I've filed an ticket to track it separately: https://github.com/zephyrproject-rtos/zephyr/issues/64675